### PR TITLE
UOE-11665: Fix-Panic for AppLovin requests if the multi-bid multi-floor feature is enabled for the profile but not configured for certain ad units

### PR DESCRIPTION
**Pull request by AvinashKapre**

### CODE REVIEW

File: modules/pubmatic/openwrap/defaultbids.go [hunk 1]
```diff
@@ -204,7 +203,10 @@ func (m *OpenWrap) addDefaultBidsForMultiFloorsConfig(rctx *models.RequestCtx, b
 	}
 
 	for impID, impCtx := range rctx.ImpBidCtx {
-		adunitFloors = rctx.AppLovinMax.MultiFloorsConfig.Config[impCtx.TagID]
+		adunitFloors, ok := rctx.AppLovinMax.MultiFloorsConfig.Config[impCtx.TagID]
```
Comment (Pubmatic-Dhruv-Sonone) [inline modules/pubmatic/openwrap/defaultbids.go:206]: Can you also add `len(adunitFloors) > 0` check here?

File: modules/pubmatic/openwrap/defaultbids.go [hunk 2]
```diff
@@ -204,7 +203,10 @@ func (m *OpenWrap) addDefaultBidsForMultiFloorsConfig(rctx *models.RequestCtx, b
...
+		if !ok {
```
Comment (Pubmatic-Dhruv-Sonone) [inline modules/pubmatic/openwrap/defaultbids.go:207]: Can we change this condition to `if !ok  || len(adunitFloors) == 0 {` ?

File: modules/pubmatic/openwrap/defaultbids_test.go [hunk 3]
```diff
@@ -335,6 +335,64 @@ func TestOpenWrap_addDefaultBidsForMultiFloorsConfig(t *testing.T) {
 				},
 			},
 		},
+		{
```
Comment (Pubmatic-Dhruv-Sonone) [inline modules/pubmatic/openwrap/defaultbids_test.go:338]: Add UT for empty floors array

**Review by Pubmatic-Dhruv-Sonone — CHANGES_REQUESTED** [review]

File: modules/pubmatic/openwrap/defaultbids_test.go (hunk 3 above)
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/defaultbids_test.go:338]: Done

**Review by Pubmatic-Dhruv-Sonone — DISMISSED** [review]

File: analytics/pubmatic/helper.go [hunk 4]
```diff
@@ -81,7 +81,7 @@ var send = func(rCtx *models.RequestCtx, url string, headers http.Header, mhc mh
 
 // RestoreBidResponse restores the original bid response for AppLovinMax from the signal data
 func RestoreBidResponse(rctx *models.RequestCtx, ao analytics.AuctionObject) error {
-	if rctx.Endpoint != models.EndpointAppLovinMax {
+	if rctx.Endpoint != models.EndpointAppLovinMax || rctx.AppLovinMax.Reject {
```
Comment (pm-jaydeep-mohite) [inline analytics/pubmatic/helper.go:84]: Can we add check for `len(ao.Response.SeatBid[0]) > 0 && len(ao.Response.SeatBid[0].Bid[0]) > 0` at line 88 ?

Comment (AvinashKapre) [reply analytics/pubmatic/helper.go:84]: Done

**Review by Pubmatic-Dhruv-Sonone — APPROVED** [review]

**Review by pm-viral-vala — APPROVED** [review]

**Pull request merged**